AutoDarkModeApp/ViewModels/SettingsViewModel.cs

```diff
@@ -197,15 +199,43 @@ private void LoadSettings()
         {
             _isInitializing = true;
 
-            var languageText = await _localSettingsService.ReadSettingAsync<string>("Language");
-            if (languageText != null)
+            var language = await _localSettingsService.ReadSettingAsync<string>("SelectedLanguageCode");
+            if (!string.IsNullOrEmpty(language))
             {
-                Language = languageText.Replace("\"", "");
+                SelectedLanguageCode = language;
             }
             else
             {
-                Language = "English (English)";
-                await _localSettingsService.SaveSettingAsync("Language", "English (English)");
+                //search for regional language
+                var preferredLanguages = ApplicationLanguages.Languages;
+                string topLanguage;
+                if (preferredLanguages.Any()) // just to be sure
+                {
+                    topLanguage = preferredLanguages.FirstOrDefault(); // e.g., "fr-FR"
+                }
+                else // extremely rare case
+                {
+                    topLanguage = CultureInfo.CurrentUICulture.Name; // e.g., "fr-FR"
+                }
+                if (LanguageConstants.SupportedCultures.Contains(topLanguage))
+                {
+                    SelectedLanguageCode = topLanguage;
+                }
+                else
+                {
+                    //search for neutral language
+                    var neutralLanguage = topLanguage.Split('-')[0]; // e.g., "fr"
+                    if (LanguageConstants.SupportedCultures.Contains(neutralLanguage))
+                    {
+                        SelectedLanguageCode = neutralLanguage;
+                    }
+                    else
+                    {
+                        SelectedLanguageCode = "en";
+                        //or default to <DefaultLanguage/>
+                    }
+                }
+                await _localSettingsService.SaveSettingAsync("SelectedLanguageCode", SelectedLanguageCode);
```

Comment:
@Spiritreader @ChenYiLins this whole block of code uses some nice checks to find the best language. I think I want to use a function/method for this, so that I can reuse that safety feature in other places. For example

```cs
// App.xaml.cs, in **OnLaunched**
 Microsoft.Windows.Globalization.ApplicationLanguages.PrimaryLanguageOverride = Something()
// GeolocatorService.cs
 _langcode uses CultureInfo
```

Can you help me on my way?

Reply:
@ChenYiLins @Spiritreader have a look at the new **GetDefaultLanguageAsync** in the latest commit. It is used in _App.xaml.cs_ and in _GeolocatorService.cs_ and it checks the data from the settings file and corrects any mistakes (like missing or invalid string, see previous comment). At the time of writing this, it lives in **SettingsViewModel**, but it should be moved to a different place...